sockeye/decoder.py

```diff
@@ -159,30 +162,31 @@ def init_state_from_encoder(self,
 
         :param encoder_outputs: Encoder outputs. Shape: (batch, source_length, encoder_dim).
         :param encoder_valid_length: Valid lengths of encoder outputs. Shape: (batch,).
-        :param is_inference: Whether to return states for inference or for training.
         :return: Initial states.
         """
         source_mask = self.valid_length_mask(encoder_outputs, encoder_valid_length)
 
-        if is_inference:
+        # (batch_size, 1)
+        step = mx.nd.expand_dims(mx.nd.zeros_like(encoder_valid_length), axis=1)
```

Comment:
Why the shape change from (1,) to (1,1)?

Reply:
because I wanted the graph to be the same for training and decoding. So the step always has the time dimension now.

Reply:
ah I see, thanks!